dl/coordinator: property to disable snapshot expiry

andrwng opened the pull request:
Adds a cluster config property to disable Iceberg snapshot expiry. This is primarily and escape hatch in case the Iceberg catalog expects to perform maintenance on its own.

<!--
See https://github.com/redpanda-data/redpanda/blob/dev/CONTRIBUTING.md#pull-request-body
for more details and examples of what is expected in a PR body.

Content in this top section is REQUIRED. Describe, in plain language, the motivation
behind the change (bug fix, feature, improvement) in this PR and how the included
commits address it.

Add the GitHub keyword `Fixes` to link to bug(s) this PR will fix, e.g.
  Fixes #ISSUE-NUMBER, Fixes #ISSUE-NUMBER, ...

If this PR is a backport, link to the original with `Backport of PR`, e.g.
  Backport of PR #PR-NUMBER
-->

## Backports Required

<!-- Checking at least one of the checkboxes is REQUIRED if this PR is not a backport. -->

- [x] none - not a bug fix
- [ ] none - this is a backport
- [ ] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [ ] v24.3.x
- [ ] v24.2.x
- [ ] v24.1.x

## Release Notes

* none
<!--
If the changes in this PR do not need to be mentioned in the release
notes, then don't add a sub-section and simply list `none`, e.g.

* none

Otherwise, adding a sub-section or `none` is REQUIRED if the PR is not a backport PR.
If this is a backport PR, adding contents to this section will override
the release notes section inherited from the original PR to dev.

Add one or more of the sub-sections with a short description bullet
point of the change, e.g.

### Bug Fixes

* Short description of the bug fix if this is a PR to `dev` branch.

### Features

* Short description of the feature. Explain how to configure.

### Improvements

* Short description of how this PR improves existing behavior.

-->

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--760d7606-ee10-465f-8fff-934debadb33e-->
<!--TEST RESULTS-->
<!--38338f70-5937-495c-bd07-d980aca04fc8-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/62494">build#62494</a></summary>


| test_id | test_kind | job_url | test_status | passed |
|:------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.cloud_storage_chunk_read_path_test.CloudStorageChunkReadTest.test_read_when_segment_size_smaller_than_chunk_size | ducktape | https://buildkite.com/redpanda/redpanda/builds/62494#01954fb2-9f80-45bd-9c58-301417aead09 | FLAKY | 1/2 |
| rptest.tests.full_disk_test.LogStorageMaxSizeSI.test_stay_below_target_size.log_segment_size=1048576.cleanup_policy=compact | ducktape | https://buildkite.com/redpanda/redpanda/builds/62494#01954fb2-9f80-45bd-9c58-301417aead09 | FLAKY | 1/3 |
| rptest.tests.scaling_up_test.ScalingUpTest.test_scaling_up_with_recovered_topic | ducktape | https://buildkite.com/redpanda/redpanda/builds/62494#01954fb6-8696-4675-926a-b29b281846e8 | FLAKY | 1/4 |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/62553">build#62553</a></summary>


| test_id                                                                                                                     | test_kind   | job_url                                                                                   | test_status   | passed   |
|:----------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.availability_test.AvailabilityTests.test_availability_when_one_node_failed                                     | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62553#01955d2c-9a41-4987-b91d-09457fde4259 | FLAKY         | 1/2      |
| rptest.tests.full_disk_test.LogStorageMaxSizeSI.test_stay_below_target_size.log_segment_size=1048576.cleanup_policy=compact | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62553#01955d2c-9a43-4533-a56f-6192b34fa1d5 | FLAKY         | 1/3      |
</details>
<!--/body-->

andrwng committed:
dl/coordinator: property to disable snapshot expiry

Adds a cluster config property to disable Iceberg snapshot expiry. This
is primarily and escape hatch in case the Iceberg catalog expects to
perform maintenance on its own.
